integration-libs/opf/components/opf-checkout-payment-wrapper/opf-checkout-payment-wrapper.service.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { ActiveCartService } from '@spartacus/cart/base/core';
+import { RoutingService, UserIdService } from '@spartacus/core';
+import { OpfResourceLoaderService } from '@spartacus/opf/core';
+import {
+  OpfCheckoutFacade,
+  OpfOtpFacade,
+  OpfPaymentMethodType,
+  OpfRenderPaymentMethodEvent,
+  PaymentSessionData,
+} from '@spartacus/opf/root';
+
+import { BehaviorSubject, Observable, combineLatest } from 'rxjs';
+import { filter, map, switchMap } from 'rxjs/operators';
+
+@Injectable()
+export class OpfCheckoutPaymentWrapperService {
+  constructor(
+    protected opfCheckoutService: OpfCheckoutFacade,
+    protected opfOtpService: OpfOtpFacade,
+    protected opfResourceLoaderService: OpfResourceLoaderService,
+    protected userIdService: UserIdService,
+    protected activeCartService: ActiveCartService,
+    protected routingService: RoutingService
+  ) {}
+
+  protected activeCartId: string;
+
+  protected renderPaymentMethodEvent$ =
+    new BehaviorSubject<OpfRenderPaymentMethodEvent>({
+      isLoading: false,
+    });
+
+  getRenderPaymentMethodEvent(): BehaviorSubject<OpfRenderPaymentMethodEvent> {
```

Comment:
IMO better to create public observable from subject with `this.renderPaymentMethodEvent$.asObservable()`.
Currently you have private subject but add method that make it public with all it's build in methods, for example right now it's possible to do `.next` to this subject from `OpfCheckoutPaymentWrapperComponent`. (`renderPaymentMethodEvent$.next()`). I think better to encapsulate this inside of service.

I'm usually doing this with creating subscription, something like 
`renderPaymentMethodEventSub = new BehaviorSubject` and after this declaring 
`renderPaymentMethodEvent$ = this.renderPaymentMethodEventSub.asObservable()`

But you could also return this `this.renderPaymentMethodEvent$.asObservable()` in `getRenderPaymentMethodEvent()` method.